rfcs/0006-picasso-runtime-upgrades.md

```diff
@@ -0,0 +1,155 @@
+
+Document is not intentended to described detailed specificaitons of quality gates.
+
+Document specifies relevant domains and stakeholders and teams, but not names exactly what communication channel these would use as communicaiton medium.
+
+## Constraints
+
+- Upgrades can be done only under `Root` origin, but cannot be `single signature`. And yet we must be able to release very fast, for examples in case of hotfix.
+
+- Upgrades are not immediate because runtime should be copied to parachain collators and relay validators and be enabled simultaneously on a specific block in the future.
+
+- There should be aligment of stakeholders on quality of release to be sure that bridged Picasso assets are secure. 
+
+
+## Legend
+
+***represent on chain things*** which can be interacted via Polkadot.js or Composable parachain SDK.
+
+## Flow
+
+### Overview of flow
+
+Relase notes collected, specifically `git commit`s which must be inlcuded into release are traceable to projects or pull request `item`s.
+
+Relevant notes and references to runtime build artifact (`wasm`) is shared into company channel with relevant representativses.
+
+Represantatives agree with this release.
+
+Release is stored as ***preimage*** on Picasso. Preimage `hash` is shared in channel with ***counci*** representatives.
+
+Council creates default YES ***motion*** to vote on enactment of premiage on democracy. Shareholdes and ***technicalCollective*** fast track enactment of new runtime.
+
+Each step executed Here is mapping of teams to roles they play in process
+
+#### Overview of actors
+
+| role | groups of people| area |  
+|-----------------|-----|-----|
+| release engineers | `@ComposableFi/sre`| | 
+| integration stakeholders| not direct contributers | depend on runtime UI/Frontend/Bots/Explorer/Data integation |
+| council |   ***concil***(on chain list of keys with attahed ***identity***) | tokenomics, marketing |
+| technical| ***technicalCollective*** | speed up on chain changes | 
+
+#### Overview of core quality gates 
+
+Minimal set of gates.
+
+All changes to runtime are forward compatible.
+Specifically, if extinsic was added, it is retained and new version are added with larger dispatch identifiers.
+If storage was added, it is migrated by relevant runtime code on upgrade to new version. Storage is not ugprade in place as there consumers which may not be able to read it after it is changed. Same relates to events.
+
+Runtime was upraded on Dali Rococo Testnet and was producing blocks here at for several hours.
+
+Commit form which release is done passed full set of relevant check on protected main branch. 
+You cannot force push. 
+Each protocol on chain has ability to stop operating as of decision of its governance without dependency on runtime upgrade.
+
+Resonable represenatives of stakeholders approved release to Piccaso.
+
+All pallets included in Picasso runtime has sufficient amount of audit.
```

Comment:
```suggestion
#### Overview of core quality gates 

Minimal set of gates:

- All changes to runtime are forward compatible.
- - If extrinsic was added, it is retained and new versions are added with larger dispatch identifiers.
- - If storage was added, it is migrated by relevant runtime code on upgrade to the new version. Storage is not upgraded in place as there are consumers which may not be able to read it after it is changed. The same relates to events.

- The runtime was upgraded on Dali Rococo Testnet and produced blocks here for several hours.

- Commit the form that validates the release as done and having passed the complete set of relevant checks on the protected main branch. 
- - You cannot force push. 
- - Each protocol on chain can stop operating as a decision of its governance without dependency on runtime upgrade.

- Reasonable representatives of stakeholders approved the release to Piccaso.

- All pallets included in Picasso runtime have been audited sufficiently.
```
Question: When you say "commit form" do you mean to commit a "release readme document"?
As I understand this is a document by the governance body and proves the release as complete and successfully checked.